src/nat/eval/llm_validator.py

```
@@ -0,0 +1,324 @@
+        ValueError: If config.llms is not properly structured.
+    """
+    # Validate config structure
+    if not hasattr(config, "llms"):
+        raise ValueError("Config object does not have 'llms' attribute")
+
+    if not config.llms:
+        logger.debug("No LLMs defined in config, skipping validation")
+        return
+
+    if not isinstance(config.llms, dict):
+        raise ValueError(f"config.llms must be a dict, got {type(config.llms)}")
```

Ruff (0.14.10)

Prefer `TypeError` exception for invalid type

(TRY004)